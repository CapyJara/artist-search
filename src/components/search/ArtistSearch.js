import React, { PureComponent } from 'react';
import SearchForm from './SearchForm';
import ArtistList from './ArtistList';
import artistGetter from '../../services/artistGetter';

export default class ArtistSearch extends PureComponent {
  state = {
    searchTerm: '',
    artistResults: []
  }

  // 1. user types in search term, hits submit. Search term is set in state.
  // 2. api call is made with searchTerm, updates state with results
  // 3. results are loaded in the artistList
  handleSubmit = event => {
    event.preventDefault();
    
    return artistGetter(this.state.searchTerm)
      .then(res => {
        console.log(res);
        this.setState({ artistResults: res.artists });
      });
  }

    handleChange = ({ target }) => {
      this.setState({ [target.name]: target.value });
    }

    render() {
      const { searchTerm } = this.state;
      return (
      <>
        <SearchForm handleChange={this.handleChange} handleSubmit={this.handleSubmit} term={searchTerm} />
        {/* <ArtistList /> */}
        <p>{searchTerm}</p>
      </>
      );
    }
  
}
