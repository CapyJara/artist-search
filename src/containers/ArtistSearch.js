import React, { PureComponent } from 'react';
import SearchForm from '../components/search/SearchForm';
import ArtistList from '../components/search/ArtistList';
import artistGetter from '../services/artistGetter';

export default class ArtistSearch extends PureComponent {
  state = {
    searchTerm: '',
    artistResults: []
  }

  // 1. user types in search term, hits submit. Search term is set in state.
  // 2. api call is made with searchTerm, updates state with results
  // 3. results are loaded in the artistList
  handleSubmit = event => {
    event.preventDefault();
    
    return artistGetter(this.state.searchTerm)
      .then(res => {
        this.setState({ artistResults: res.artists });
      });
  }

    handleChange = ({ target }) => {
      this.setState({ [target.name]: target.value });
    }

    render() {
      const { searchTerm, artistResults } = this.state;
      return (
      <>
        <SearchForm handleChange={this.handleChange} handleSubmit={this.handleSubmit} term={searchTerm} />
        <ArtistList artistArr={artistResults} />
      </>
      );
    }
  
}
