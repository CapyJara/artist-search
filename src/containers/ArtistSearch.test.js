import React from 'react';
import { shallow } from 'enzyme';
import ArtistSearch from './ArtistSearch';

describe('ArtistSearch component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<ArtistSearch />);
    expect(wrapper).toMatchSnapshot();
  });
});
