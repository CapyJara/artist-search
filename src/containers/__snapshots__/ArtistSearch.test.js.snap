// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistSearch component matches snapshot 1`] = `
<Fragment>
  <SearchForm
    handleChange={[Function]}
    handleSubmit={[Function]}
    term=""
  />
  <ArtistList
    artistArr={Array []}
  />
  <Paging
    changePage={[Function]}
    currentPage={0}
    totalPages={1}
  />
</Fragment>
`;
